example/morphology.cpp

```diff
@@ -0,0 +1,130 @@
+//
+// Copyright 2021 Prathamesh Tagore <prathameshtagore@gmail.com>
+//
+// Use, modification and distribution are subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#include <boost/gil/extension/io/png.hpp>
+#include <boost/gil/image_processing/morphology.hpp>
+#include <vector>
+#include<iostream>
+#include<map>
+#include<string>
+//Default structuring element is SE = [1,1,1]
+//                                    |1,1,1|
+//                                    [1,1,1]
+//SE(1,1)(center pixel) is the one which coincides with the currently considered pixel of the 
+//image to be convolved. The structuring element can be easily changed by the user.
+
+int main(int argc,char **argv)
+{
+    std::map<std::string,bool>operations;
+    if(argc < 4 || argc > 11)
+    {
+        std::cout<<"Wrong format of command line arguments.\n";
+        std::cout<<"Correct format is <Input_image.png> <Output_image_template>"
```

Comment:
you can omit semicolon on previous line and `std::cout` here, so it will continue streaming.

Reply:
I'd suggest to `throw std::invalid_argument` instead of `std::cout`